packages/mobx-state-tree/__tests__/core/this.test.ts

```diff
@@ -0,0 +1,85 @@
+import { types } from "../../src"
+import { isObservableProp, isComputedProp } from "mobx"
+
+test("this support", () => {
+    const M = types
+        .model({ x: 5 })
+        .views(self => ({
+            get x2() {
+                return self.x * 2
+            },
+            get x4() {
+                return this.x2 * 2
+            },
+            boundTo() {
+                return this
+            },
+            innerBoundTo() {
+                return () => this
+            },
+            isThisObservable() {
+                return (
+                    isObservableProp(this, "x2") &&
+                    isObservableProp(this, "x4") &&
+                    isObservableProp(this, "localState") &&
+                    isComputedProp(this, "x2")
+                )
+            }
+        }))
+        .volatile(self => ({
+            localState: 3,
+            getLocalState() {
+                return this.localState
+            },
+            getLocalState2() {
+                return this.getLocalState() * 2
+            }
+        }))
+
+        .actions(self => {
+            return {
+                xBy(by: number) {
+                    return self.x * by
+                },
+                setX(x: number) {
+                    self.x = x
```

Comment:
Just to double check, `this.x` won't work, correct? `this` revers to the definition and not to the instance? I would make sure to mention that at the place in the readme where you introduce the `this` workaround in the Readme, otherwise people will assume it refers to the same thing as `self`

Really smart solution btw!

Reply:
actually I think it would work, but TS would complain
I can add a unit test for it

of course that depends on the wanted behaviour (which depends on the other line of code you commented on)